core/services/mongodb-remote/src/main/java/com/mongodb/stitch/core/services/mongodb/remote/sync/internal/DataSynchronizer.java

```diff
@@ -1703,14 +1702,17 @@ public void syncDocumentsFromRemote(
       final BsonValue... documentIds
   ) {
     this.waitUntilInitialized();
+    boolean added = false;
 
     try {
       ongoingOperationsGroup.enter();
       for (final BsonValue documentId : documentIds) {
-        syncConfig.addSynchronizedDocument(namespace, documentId);
+        added = added || syncConfig.addSynchronizedDocument(namespace, documentId);
```

Comment:
[bug] I think short-circuit boolean logic means that if `added` is true, then the document won't be added to the syncConfig, meaning that if you try to sync multiple ids, it will ignore all the ids after the first added one.

I think this should be 

```java
final boolean docAdded = syncConfig.addSynchronizedDocument(namespace, documentId);
added = added || docAdded;
```

you could also probably rename `added` to `anyDocsAdded` or `someDocsAdded`

Reply:
Oh good catch I need non-short circuiting behavior